src/libstore/build/local-derivation-goal.cc

```diff
@@ -1672,6 +1673,12 @@ void setupSeccomp()
         if (seccomp_rule_add(ctx, SCMP_ACT_ERRNO(EPERM), SCMP_SYS(fchmodat), 1,
                 SCMP_A2(SCMP_CMP_MASKED_EQ, (scmp_datum_t) perm, (scmp_datum_t) perm)) != 0)
             throw SysError("unable to add seccomp rule");
+
+#if defined(NIX_SYSCALL_FCHMODAT2)
```

Comment:
you shouldn't need this guard i don't believe. (or, rather, please make this guard unnecessary)

Reply:
Good point, this is also guarded by __linux__/HAVE_SECCOMP 👍 
(And yes, generally a good thing to not have this silently removed by accident)